tsl/src/continuous_aggs/materialize.c

```diff
@@ -520,6 +564,73 @@ drain_invalidation_log(int32 raw_hypertable_id, List **invalidations_out)
 						&scan_state /*=data*/);
 }
 
+/* NOTE: gets invalidations from materialization_invalidation_log table */
+static void
+drain_materialization_invalidation_logs(int32 cagg_id, List **invalidations_out)
+{
+	InvalidationScanState scan_state = {
+		.invalidations = invalidations_out,
+		.mctx = CurrentMemoryContext,
+	};
+	ScanKeyData scankey[1];
+
+	ScanKeyInit(&scankey[0],
+				Anum_continuous_aggs_materialization_invalidation_log_idx_materialization_id,
+				BTEqualStrategyNumber,
+				F_INT4EQ,
+				Int32GetDatum(cagg_id));
+
+	ts_catalog_scan_all(CONTINUOUS_AGGS_MATERIALIZATION_INVALIDATION_LOG /*=table*/,
+						CONTINUOUS_AGGS_MATERIALIZATION_INVALIDATION_LOG_IDX /*=indexid*/
+						,
+						scankey /*=scankey*/,
+						1 /*=num_keys*/,
+						scan_del_materialization_invalidation_tuple /*=tuple_found*/,
```

Comment:
why can we not use `scan_take_invalidation_tuple` again?

Reply:
different catalog table.

Reply:
🤔 it's still the same logic, but they are _technically_ different structs, though the layout of the tables should always be the same. I am very conflicted on this one

Reply:
I  strongly think we should not reuse the structs. It should be separate functions.